### DISCUSSION TITLE
Add exclude key for directories

### DISCUSSION THREAD

**Discussion by buhruf** [BetaHuhn/repo-file-sync-action#26] [Ideas] [answered]
Hi there, 

It looks like you're action has really been making some progress in the last few months. I've been using it for some file sync tasks and it's been working splendidly. I did have one idea that might be useful: an **exclude** key for the sync directories action:

```
user/repo:
  - source: workflows/
    dest: .github/workflows/
    exclude:  
```

I've found that I need to spend a bit more time listing the files I DO want to sync to avoid the files I don't, when really an enhanced directory sync where I could declare the source and destination repo and then list the one or two files I DON'T want synced would be very economical. 

Anyways. Just an idea. Thanks again for the great work you're doing here.

Best,

Derek

**Comment by BetaHuhn** [comment]
Hey @buhruf,

First of all, thanks for the nice words! It's always nice to hear back from someone using one of my Actions.

 > I did have one idea that might be useful: an exclude key for the sync directories action

Yeah that's a great idea! I haven't had the need for this myself, but I can totally see a use case for this.

**Reply by BetaHuhn** [reply]
I implemented this on the [develop](https://github.com/BetaHuhn/repo-file-sync-action/tree/develop) branch. I also added an [example](https://github.com/BetaHuhn/repo-file-sync-action/tree/develop#exclude-certain-files-when-syncing-directories) to the README which shows how you can use the `exclude` key.

For this to work I had to change the library which is used to copy files internally, which should also fix #15.

You can try it out by using the Action like this:

```yml
uses: BetaHuhn/repo-file-sync-action@develop
```

@buhruf  Let me know if everything works for you and I will merge this into master and release a new version 😄

**Reply by buhruf** [reply]
Thank you kindly, @BetaHuhn! I'll test this in detail the first chance I get on Monday and get back to you with my findings. 

Again, I really appreciate the work you're doing on this Action. It just keeps getting better and better 😄

**Comment by buhruf** [comment]
Hi @BetaHuhn,

The changes you deployed to `develop` are working as expected.  🎉  Thanks again for your work here! This is such a useful enhancement and is going to dramatically simplify my efforts! 

My hat goes off to you again. Well done! 🎉

**Reply by BetaHuhn** [reply]
Great!

> Thanks again for your work here! This is such a useful enhancement and is going to dramatically simplify my efforts!

No problem 😄 

Note: If you depend on this Action or use it in a commercial project, maybe _consider_ supporting me on **[Ko-fi](https://ko-fi.com/betahuhn)** or via **[PayPal](https://paypal.me/payschiller).** This helps me stay motivated to keep maintaining/improving this Action and all my other projects.

**Comment by BetaHuhn** [comment, marked as answer]
The `exclude` key feature is now a part of [v1.5.0](https://github.com/BetaHuhn/repo-file-sync-action/releases/tag/v1.5.0) 🎉

You can use it by either specifying the `v1` or the `latest` tag:

```yml
uses: BetaHuhn/repo-file-sync-action@latest
```

And then in your `sync.yml`:

```yml
user/repo:
  - source: workflows/
    dest: .github/workflows/
    exclude: |
      node.yml
      lint.yml
```

> **Note:** the exclude file path is relative to the source path

**Reply by buhruf** [reply]
🎉 Fantastic work as always, @BetaHuhn! Well done!
